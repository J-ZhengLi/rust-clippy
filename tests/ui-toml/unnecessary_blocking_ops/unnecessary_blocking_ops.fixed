#![warn(clippy::unnecessary_blocking_ops)]
use std::thread::sleep;
use std::time::Duration;
use std::{fs, io};

mod async_mod {
    pub async fn sleep(_dur: std::time::Duration) {}
    pub async fn read_to_string(mut reader: std::io::Stdin) -> Result<String, ()> {
        Ok(String::new())
    }
}

mod blocking_mod {
    pub async fn sleep(_dur: std::time::Duration) {}
}

pub async fn async_fn() {
    crate::async_mod::sleep(Duration::from_secs(1)).await;
    //~^ ERROR: blocking function call detected in an async body
    crate::async_mod::sleep(Duration::from_secs(1)).await;
    //~^ ERROR: blocking function call detected in an async body
    let mut r: &[u8] = b"hello";
    let mut w: Vec<u8> = vec![];
    tokio::io::copy(&mut r, &mut w).await.unwrap();
    //~^ ERROR: blocking function call detected in an async body
    let _cont = crate::async_mod::read_to_string(io::stdin()).await.unwrap();
    //~^ ERROR: blocking function call detected in an async body
    fs::create_dir("").unwrap(); // Don't lint, config overrided
}

fn main() {}
