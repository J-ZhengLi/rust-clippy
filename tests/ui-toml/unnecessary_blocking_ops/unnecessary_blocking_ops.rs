//@no-rustfix
#![warn(clippy::unnecessary_blocking_ops)]
use std::thread::sleep;
use std::time::Duration;
use std::{fs, io};

mod async_mod {
    pub async fn sleep(_dur: std::time::Duration) {}
    pub async fn read_to_string(mut reader: std::io::Stdin) -> Result<String, ()> {
        Ok(String::new())
    }
}

mod blocking_mod {
    pub async fn sleep(_dur: std::time::Duration) {}
}

pub async fn async_fn() {
    sleep(Duration::from_secs(1));
    //~^ ERROR: blocking function call detected in an async body
    fs::remove_dir("").unwrap();
    //~^ ERROR: blocking function call detected in an async body
    fs::copy("", "_").unwrap();
    //~^ ERROR: blocking function call detected in an async body
    let mut r: &[u8] = b"hello";
    let mut w: Vec<u8> = vec![];
    io::copy(&mut r, &mut w).unwrap();
    //~^ ERROR: blocking function call detected in an async body
    let _cont = io::read_to_string(io::stdin()).unwrap();
    //~^ ERROR: blocking function call detected in an async body
    fs::create_dir("").unwrap();
    //~^ ERROR: blocking function call detected in an async body
}

fn main() {}
