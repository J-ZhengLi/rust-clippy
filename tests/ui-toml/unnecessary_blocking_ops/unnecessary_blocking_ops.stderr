error: blocking function call detected in an async body
  --> tests/ui-toml/unnecessary_blocking_ops/unnecessary_blocking_ops.rs:18:5
   |
LL |     sleep(Duration::from_secs(1));
   |     ^^^^^------------------------
   |     |
   |     help: try using its async counterpart: `crate::async_mod::sleep(Duration::from_secs(1)).await`
   |
   = note: `-D clippy::unnecessary-blocking-ops` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::unnecessary_blocking_ops)]`

error: blocking function call detected in an async body
  --> tests/ui-toml/unnecessary_blocking_ops/unnecessary_blocking_ops.rs:20:5
   |
LL |     blocking_mod::sleep(Duration::from_secs(1));
   |     ^^^^^^^^^^^^^^^^^^^------------------------
   |     |
   |     help: try using its async counterpart: `crate::async_mod::sleep(Duration::from_secs(1)).await`

error: blocking function call detected in an async body
  --> tests/ui-toml/unnecessary_blocking_ops/unnecessary_blocking_ops.rs:24:5
   |
LL |     io::copy(&mut r, &mut w).unwrap();
   |     ^^^^^^^^----------------
   |     |
   |     help: try using its async counterpart: `tokio::io::copy(&mut r, &mut w).await`

error: blocking function call detected in an async body
  --> tests/ui-toml/unnecessary_blocking_ops/unnecessary_blocking_ops.rs:26:17
   |
LL |     let _cont = io::read_to_string(io::stdin()).unwrap();
   |                 ^^^^^^^^^^^^^^^^^^-------------
   |                 |
   |                 help: try using its async counterpart: `crate::async_mod::read_to_string(io::stdin()).await`

error: aborting due to 4 previous errors

